x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curations_logic.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+
+import {
+  clearFlashMessages,
+  setSuccessMessage,
+  flashAPIErrors,
+} from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { DELETE_MESSAGE, SUCCESS_MESSAGE } from './constants';
+import { Curation, CurationsAPIResponse } from './types';
+
+interface CurationsValues {
+  dataLoading: boolean;
+  curations: Curation[];
+  meta: Meta;
+}
+
+interface CurationsActions {
+  onCurationsLoad(response: CurationsAPIResponse): CurationsAPIResponse;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
+  loadCurations(): void;
+  deleteCurationSet(id: string): string;
+}
+
+export const CurationsLogic = kea<MakeLogicType<CurationsValues, CurationsActions>>({
```

Comment:
I only added the current values/actions needed for the overview table for this PR.

This is intentional: I plan on deviating from the current ent-search repo in that I plan on splitting up the old `CurationsLogic` into `curations_logic.ts` and `curation_logic.ts` (i.e., separating the `curations` collection from individual `curation` pages). The current CurationsLogic file in ent-search feels unwieldy/difficult to parse to me and I can only imagine what the unit tests for that file will look like. I strongly feel separating out the two concerns will help with both migration and readability.

The next curation-specific PR (create curation view) will probably be the last PR that adds to CurationsLogic, and after that future PRs will be creating/adding on to a new CurationLogic file.

Reply:
I think that's a good call.

Something that *may* help, not sure if it will be an issue or not, but we ran into this with EngineLogic.

We may want to consider "key"-ing CurationLogic and using `BindLogic`. It's a feature available in the latest version of Kea: https://github.com/keajs/kea/issues/120.

Again, not sure if it will be an issue, so you may not want to cross that bridge if you don't have to, but thought I would mention it as something to look out for.

Reply:
This is super helpful, thanks Jason!